src/PotentialSimulation/ConvergenceAndRefinement.jl

```diff
@@ -88,282 +88,181 @@ function _update_till_convergence!( fssrb::PotentialSimulationSetupRB{T, N1, N2}
+end
 
-                # x
-                if ix != size(potential, 1)
-                    if abs(potential[ix + 1, iy, iz] - potential[ix, iy, iz]) >= refinement_value_x
-                        if ax_x[ix + 1] - ax_x[ix] > minimum_distance_x
-                            if !in(ix, inds_x) push!(inds_x, ix) end
-                        end
-                    end
-                end
+_convert_closed_axis(::Type{DiscreteAxis{T, BL, BR, I}}, ax::DiscreteAxis{T, BL, BR, I}) where {T, BL, BR, I} = ax
 
-                # y
-                if iy != size(potential, 2)
-                    if abs(potential[ix, iy + 1, iz] - potential[ix, iy, iz]) >= refinement_value_y
-                        if ax_y[iy + 1] - ax_y[iy] > minimum_distance_y
-                            if !in(iy, inds_y) push!(inds_y, iy) end
-                        end
-                    end
-                end
+function _convert_closed_axis(::Type{DiscreteAxis{T, BL, BR, Interval{:closed, :open, T}}}, ax::DiscreteAxis{T, BL, BR, ClosedInterval{T}}) where {T, BL, BR}
+    DiscreteAxis{T, BL, BR, Interval{:closed, :open, T}}(Interval{:closed, :open, T}(ax.interval.left, ax.interval.right), ax.ticks[1:end-1])
+end
+function _convert_closed_axis(::Type{DiscreteAxis{T, BL, BR, Interval{:open, :closed, T}}}, ax::DiscreteAxis{T, BL, BR, ClosedInterval{T}}) where {T, BL, BR}
+    DiscreteAxis{T, BL, BR, Interval{:open, :closed, T}}(Interval{:open, :closed, T}(ax.interval.left, ax.interval.right), ax.ticks[2:end])
+end
 
-                # z
-                if iz != size(potential, 3)
-                    if abs(potential[ix, iy, iz + 1] - potential[ix, iy, iz]) >= refinement_value_z
-                        if ax_z[iz + 1] - ax_z[iz] > minimum_distance_z
-                            if !in(iz, inds_z) push!(inds_z, iz) end
-                        end
-                    end
-                end
+function _reduce_closed_values(axTypes::Tuple, a::Array{T, 3}) where {T} 
+   inds = broadcast(i -> _get_ind_range_of_closed_axis(size(a, i), axTypes[i]), (1, 2, 3))
+   a[inds...]
+end
 
+_get_ind_range_of_closed_axis(l::Int, ::Type{DiscreteAxis{T, BL, BR, ClosedInterval{T}}}) where {T, BL, BR} = 1:l
+_get_ind_range_of_closed_axis(l::Int, ::Type{DiscreteAxis{T, BL, BR, Interval{:closed, :open, T}}}) where {T, BL, BR} = 1:(l-1)
+_get_ind_range_of_closed_axis(l::Int, ::Type{DiscreteAxis{T, BL, BR, Interval{:open, :closed, T}}}) where {T, BL, BR} = 2:l
+
+
+function _create_refined_grid(p::ScalarPotential{T,3}, max_diffs::NTuple{3, T}, minimum_distances::NTuple{3, T}) where {T}
+    n_1 = Int.(floor.([maximum(abs.(p.data[i+1,:,:] .- p.data[i,:,:])) for i in 1:size(p.data, 1)-1] ./ max_diffs[1], digits=0)) 
+    n_2 = Int.(floor.([maximum(abs.(p.data[:,i+1,:] .- p.data[:,i,:])) for i in 1:size(p.data, 2)-1] ./ max_diffs[2], digits=0)) 
+    n_3 = Int.(floor.([maximum(abs.(p.data[:,:,i+1] .- p.data[:,:,i])) for i in 1:size(p.data, 3)-1] ./ max_diffs[3], digits=0)) 

```

Comment:
`Int.(floor.(..., digits = 0))` is just `floor.(Int, ...)`?

Reply:
Done.